utils/check_copies.py

```diff
@@ -137,6 +137,63 @@ def _should_continue(line: str, indent: str) -> bool:
     return line.startswith(indent) or len(line.strip()) == 0 or _is_definition_header_ending_line(line)
 
 
+def _sanity_check_splits(splits_1, splits_2, is_class):
+    """Check the two (inner) block structures of the corresponding code block given by `split_code_into_blocks` match.
+
+    For the case of `class`, a consecutive sequence of blocks which names (e.g. inner class/func/method) is considered
+    as `a single block with name` in this check, because we want to allow new inner blocks in one of the 2 classes.
+    However, it must be one of the following 3 cases:
+
+        - a single block without name:
+
+            class foo:
+                a = 1
+
+        - a consecutive sequence of blocks with name
+
+            class foo:
+
+                def f(x):
+                    return x
+
+        - a block without name, followed by a consecutive sequence of blocks with name
+
+            class foo:
+                a = 1
+
+                def f(x):
+                    return x
+
+                def g(x):
+                    return None
+
+    For the case of `function or method`, we don't **merge** a consecutive sequence of blocks which names:
+    the 2 code blocks must contain exactly the same structure. However, we don't require that structure to match a
+    pre-defined structure.
+    """
+    block_names_1 = []
+    block_names_2 = []
+
+    for block in splits_1[1:]:
+        if block[0].startswith("_block_without_name_"):
+            block_names_1.append("block_without_name")
+        elif not block[0].startswith("_empty_block_") and (not is_class or len(block_names_1) == 0 or block_names_1[-1].startswith("block_without_name")):
+            block_names_1.append("block_with_name")
+
+    for block in splits_2[1:]:
+        if block[0].startswith("_block_without_name_"):
+            block_names_2.append("block_without_name")
+        elif not block[0].startswith("_empty_block_") and (not is_class or len(block_names_2) == 0 or block_names_2[-1].startswith("block_without_name")):
+            block_names_2.append("block_with_name")
+
+    if is_class:
+        if not block_names_1 in [["block_without_name"], ["block_with_name"], ["block_without_name", "block_with_name"]]:
+            raise ValueError("For a class, it must have a specific structure. See the docstring of `_sanity_check_splits` in the file `utils/check_copies.py`")
```

Comment:
We can put a link to that docstring, but I can only do this once the PR is merged to `main`